zoneminder/monitor.py

```diff
@@ -81,11 +83,17 @@ def name(self) -> str:
         """Get the name of this Monitor."""
         return self._name
 
+    def update_monitor(self):
+        """Update the monitor and monitor status from the ZM server."""
+        result = self._client.get_state(self._monitor_url)
+        self._raw_result = result['monitor']
+
     @property
     def function(self) -> MonitorState:
         """Get the MonitorState of this Monitor."""
-        return MonitorState(self._client.get_state(
-            self._monitor_url)['monitor']['Monitor']['Function'])
+        self.update_monitor()
```

Comment:
I agree that having HASS call `.update_monitor()` would make more sense, but I'm not sure about breaking the API. Especially since there's no formal release notes/documentation yet. Let's keep it as you've written for now.